src/ext/graphql/transmodelapi/schema.graphql

```diff
@@ -1753,3 +1755,23 @@ input TriangleFactors {
   "How important is time expressed as a fraction of 1. Note that what this really optimises for is distance (even if that means going over terrible surfaces, so I might be slower than the safe route)."
   time: Float!
 }
+
+"A collection of selectors for what lines/trips should be included in / excluded from search"
+input TripFilterInput {
+  "A list of selectors for what lines/trips should be excluded during the search. If line/trip matches with at least one selector it will be excluded."
+  not: [TripFilterSelectInput]
+  "A list of selectors for what lines/trips should be allowed during search. I order to be accepted a trip/line has to match with at least one selector.An empty list means that everything should be allowed. "
+  select: [TripFilterSelectInput]
```

Comment:
```suggestion
  not: [TripFilterSelectInput!]
  "A list of selectors for what lines/trips should be allowed during search. I order to be accepted a trip/line has to match with at least one selector.An empty list means that everything should be allowed. "
  select: [TripFilterSelectInput!]
```
To make the elements noneNull you only need to use `type(new GraphQLNonNull(...))`

Reply:
done